docs/fastdds/use_cases/rosbag_capture/rosbag_capture.rst

```diff
@@ -100,72 +108,140 @@ generators get called on your IDL files.
      ${idl_files}
    )
 
-Similarly, we need to add the ROS generators to the package.xml file so colcon can keep track of this
-dependency. Add the following to the package.xml file after the ament_cmake build_depend tags:
+Similarly, the ``package.xml`` file should be modified adding the ROS 2 TypeSupport generator dependency.
+Add the following lines to the ``package.xml`` file after the ``buildtool_depend`` tags:
 
 .. code-block:: bash
 
   <buildtool_depend>rosidl_default_generators</buildtool_depend>
   <exec_depend>rosidl_default_runtime</exec_depend>
   <member_of_group>rosidl_interface_packages</member_of_group>
 
-The last step would be to build the package. From the new_typesupport folder, the command to run is:
+The last step would be to build the package.
+Run the following command within the ``fastdds_record_typesupport`` folder:
 
 .. code-block:: bash
 
-   colcon build
+   RMW_IMPLEMENTATION=rmw_fastrtps_cpp colcon build
+
+The build process will create inside the install folder a new ROS 2 overlay with all the required
+libraries and scripts for the ROS 2 applications to use te type defined in the IDL file.
 
-The build process will create inside the install folder a new ROS2 overlay with all the required libraries
-and scripts for ROS2 applications to use your new type.
+Fast DDS Application tuning
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Fast DDS Application required modifications
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+ROS2 adds special tokens to the topic names depending on the ROS subsystem the topic belongs to. More information
+on this topic can be found `on ROS 2 design documentation
+<https://design.ros2.org/articles/topic_and_service_names.html#examples-of-ros-names-to-dds-concepts>`_ .
 
-.. warning::
-    This section is still under work.
+Using the same IDL files used earlier Fast DDS-Gen can generate the required code to handle the new type in
+Fast DDS. The required changes to make so rosbag can see your application are going to be illustrated via the
+Publisher/Subscriber example generated automatically from an IDL using Fast DDS Gen. An in-depth guide to Fast DDS-Gen
+can be found `here <https://fast-dds.docs.eprosima.com/en/latest/fastdds/dds_layer/topic/fastddsgen/fastddsgen.html>`_.
 
-Using the same IDL files used earlier Fast-DDS-Gen can generate the required code to handle the new type in
-Fast DDS. Generate this code by running:
+In the case of plain topics, the namespace "rt/" is added to all topic names so it needs to be added.
+DataType names for this generated types are structured like (using the previous IDL as example)
+"fastdds_record_typesupport::idl::HelloWorld".
```

Comment:
```suggestion
In case of plain topics, the namespace "rt/" is added by ROS 2 to the DDS topic name.
DataType names for ROS 2 generated types are structured concatenating the modules names.
For the IDL being used in this example the data type name would be "fastdds_record_typesupport::idl::HelloWorld".
```